torch/distributions/constraints.py

```diff
@@ -166,21 +168,45 @@ class _LowerTriangular(Constraint):
     Constrain to lower-triangular square matrices.
     """
     def check(self, value):
-        return (torch.tril(value) == value).min(-1)[0].min(-1)[0]
+        masked_value = value*torch.tril(value.new(*value.shape[-2:]).fill_(1.0))
```

Comment:
That should be unnecessary. `torch.tril` will slice out only the part above the diagonal from `value`

Reply:
Does that work on batches of matrices? If I call `torch.tril(torch.randn(3,2,2))` then I get a `RuntimeError: invalid argument 1: expected a matrix`…